libraries/stdlib/osl/stdosl.h

```diff
@@ -573,6 +573,237 @@ closure color volume_matte() BUILTIN;
+// -------------------------------------------------------------//
+​
+// Constructs an EDF emitting light uniformly in all directions.
+//
+//  \param  emittance   Radiant emittance of light leaving the surface.
+//  \param  label       Optional string parameter to name this component. For use in AOVs / LPEs.
+//
+closure color uniform_edf(color emittance) BUILTIN;
+​
+​
+// -------------------------------------------------------------//
+// VDF closures                                                 //
+// -------------------------------------------------------------//
+​
+// Constructs a VDF scattering light for a general participating medium, based on the Henyey-Greenstein
+// phase function. Forward, backward and uniform scattering is supported and controlled by the anisotropy input.
+//
+//  \param  albedo      Volume single-scattering albedo.
+//  \param  extinction  Volume extinction coefficient.
+//  \param  anisotropy  Scattering anisotropy [-1,1]. Negative values give backwards scattering, positive values give forward scattering, 
+//                      and 0.0 gives uniform scattering.
+//  \param  label       Optional string parameter to name this component. For use in AOVs / LPEs.
+//
+closure color anisotropic_vdf(color albedo, color extinction, float anisotropy) BUILTIN;
+
+// Constructs a VDF for light passing through a dielectric homogenous medium, such as glass or liquids.
+// The parameters transmission_depth and transmission_color control the extinction coefficient of the medium
+// in and artist-friendly way. A priority can be set to determine the ordering of overlapping media.
+//
+//  \param  albedo              Single-scattering albedo of the medium.
+//  \param  transmission_depth  Distance travelled inside the medium by white light before its color becomes transmission_color by Beer's law.
+//                              Given in scene length units, range [0,infinity). Together with transmission_color this determines the extinction
+//                              coefficient of the medium.
+//  \param  transmission_color  Desired color resulting from white light transmitted a distance of 'transmission_depth' through the medium.
+//                              Together with transmission_depth this determines the extinction coefficient of the medium.
+//  \param  anisotropy          Scattering anisotropy [-1,1]. Negative values give backwards scattering, positive values give forward scattering, 
+//                              and 0.0 gives uniform scattering.
+//  \param  ior                 Refraction index of the medium.
+//  \param  priority            Priority of this medium (for nested dielectrics).
+//  \param  label               Optional string parameter to name this component. For use in AOVs / LPEs.
+//
+closure color medium_vdf(color albedo, float transmission_depth, color transmission_color, float anisotropy, float ior, int priority) BUILTIN;
```

Comment:
@portsmouth @aconty With the use of optional params for ior and priority, as well as the alternative of using transmission_depth/transmission_color to control the extinction coefficient, I felt it was justified to instead break this out to a separate VDF for homogenous media. Let me know what you think?

Reply:
Yeah, that makes sense.

Reply:
Hm, it seems that `anisotropic_vdf` is basically specifying a (homogeneous) volumetric medium, and `medium_vdf` is doing that plus specifying IOR and priority (of the "embedding" dielectric medium), except with a different way of parameterizing the extinction coefficient..  Wouldn't it be easier to just have one `vdf` for the volume properties (maybe using the transmission parameterization, which is nicer than raw extinction),  which you pass into the full `medium_vdf` to define the general case where the volume is embedded in dielectric?

Reply:
To my mind `anisotropic_vdf` is a closure for general participating media (both homogenous and heterogeneous), whereas `media_vdf` is a specialization for homogenous dielectric media. So a specialization for a common case, where it more efficiently can handle strictly homogenous media, as well as extra logic for priority etc. with nested dielectrics. It's similar to how we have `subsurface_bssrdf` as a separate closure specialized for that use case.

I suppose we could change the parameterization of `anisotropic_vdf` to use the transmission parameterization instead of raw extinction here as well, if that makes sense for this more general case?

However I'm not sure about removing these parameters from `media_vdf` to replace them with a closure input (for plugging in an anisotropic_vdf, if I understood your proposal correctly). To my mind this would make `media_vdf` harder to implement and more confusing to use for OSL programmers as it allows plugging any closure into this input.